### ISSUE TITLE
Can this be used for both shortening and wildcard redirects to another domain?

### ISSUE THREAD

**Issue by mvs12** [YOURLS/YOURLS#2403]
Hi there,

I've just installed YOURLS and I'm testing it out to see if it would be useful my situation. The idea is certainly interesting. The main question I have is if this issue about simultaneously using the domain name as an alias was ever acted on:

[How to use YOURLS along with domain alias configuration?
](https://www.dreamincode.net/forums/topic/360985-how-to-use-yourls-along-with-domain-alias-configuration/)

At least I think that represents what I had in mind. Perhaps not, I'm not completely sure.

Basically I'm looking to see if I can do the following.
- If the shortlink exists, redirect to the long version of the link. Or just go to the page if it's an existing page like the backend.
- If the shortlink doesn't exist and the page is otherwise non existent, do a wildcard redirect to the respective page of the long version of the domain name.

I hope I've explained myself well enough, and if not I'd be happy to try again.
Regardless, thanks to the Dev for making such a project available.

**Comment by LeoColomb** [comment]
Hi @mvs12!

> If the shortlink exists, redirect to the long version of the link. Or just go to the page if it's an existing page like the backend.

This is the main feature of YOURLS, it should not be a problem.

> If the shortlink doesn't exist and the page is otherwise non existent, do a wildcard redirect to the respective page of the long version of the domain name.

This a feature that can be implemented with a plugin, as a 'not found' fallback.
See the plugin [keyword-not-found-set-field-yourls](https://github.com/8thwall/keyword-not-found-set-field-yourls/blob/master/plugin.php) for an example and the [plugin list](https://github.com/YOURLS/awesome-yourls) to find other examples.

If you publish your plugin, feel free to request addition on the list! 👍
